GDJS/scripts/lib/build-esm-library-types.js

```diff
@@ -0,0 +1,140 @@
+const ts = require('typescript');
+const path = require('path');
+const { writeFile } = require('fs/promises');
+const { readFileSync } = require('fs');
+
+/** @param {string[]} str */
+const relativePath = (...str) => path.relative(...str).replace(/\\/g, '/');
+
+const gdevelopRootPath = path.join(__dirname, '..', '..', '..');
+const gdevelopTypesDirectory = path.join(
+  gdevelopRootPath,
+  'GDJS',
+  'Runtime',
+  'types',
+  'auto-generated'
+);
+
+/**
+ * Generates a bundled dts for a module using the TypeScript programmatic API.
+ * @param {string} pathToModule The path to the module to generate a dts from.
+ * @returns {Promise<[string, string]>} The DTS file contents as plain text.
+ */
+const generateDTS = (pathToModule) => {
+  const mainModuleDirectoryName = path.dirname(pathToModule);
+  const modulePrefix = path.basename(pathToModule, '.ts');
+  const expectedMainModuleOutputName = pathToModule
+    .replace('.ts', '.d.ts')
+    .replace(/\\/g, '/');
+
+  /**
+   * @param {string} dts
```

Comment:
Can you use a name describing what is the rewrite doing? "rewriteImportdWithXxx".
So that we don't need to read the code to understand what this is doing.